### ISSUE TITLE
Missing or incorrect debug infos for placeholder variables (P2169) 

### ISSUE THREAD

**Issue by cor3ntin** [llvm/llvm-project#64430]
We added support for [P2169](https://wg21.link/P2169) ( placeholder with no name) https://reviews.llvm.org/D153536
However, this lacks debugger support and it would be nice to improve the situation before Clang 18.

In some cases the dwarf information does not contain the information for multiple placeholder, or they are reported at the same offset.

```cpp
struct t1 {
  int _;
  int _;
};
t1 v1;
```

```
0x0000002e:   DW_TAG_structure_type
                DW_AT_calling_convention        (DW_CC_pass_by_value)
                DW_AT_name      ("t1")
                DW_AT_byte_size (0x08)
                DW_AT_decl_file ("/usr/local/google/home/blaikie/dev/scratch/unused_member.cpp")
                DW_AT_decl_line (1)

0x00000034:     DW_TAG_member
                  DW_AT_name    ("_")
                  DW_AT_type    (0x00000047 "int")
                  DW_AT_decl_file       ("/usr/local/google/home/blaikie/dev/scratch/unused_member.cpp")
                  DW_AT_decl_line       (2)
                  DW_AT_data_member_location    (0x00)

0x0000003d:     DW_TAG_member
                  DW_AT_name    ("_")
                  DW_AT_type    (0x00000047 "int")
                  DW_AT_decl_file       ("/usr/local/google/home/blaikie/dev/scratch/unused_member.cpp")
                  DW_AT_decl_line       (2) 
                  DW_AT_data_member_location    (0x00)  
```

The third `DW_AT_decl_line` should be 3, not 2.

When this is fixed, we should update the release notes of clang which currently mention this deficiency

**Comment by cor3ntin** [comment]
@dwblaikie @hubert-reinterpretcast

**Comment by hubert-reinterpretcast** [comment]
Also affects lambda init-captures (also represented by `DW_TAG_member`) and, when declared on the same line, local variables.

**Comment by hubert-reinterpretcast** [comment]
Looks like this is not a front-end issue. `llvm::DIBuilder` may have interface/implementation problems.
